waku/v2/protocol/filter/server.go

```diff
@@ -317,15 +311,8 @@ func (wf *WakuFilterFullNode) pushMessage(ctx context.Context, peerID peer.ID, e
 
 // Stop unmounts the filter protocol
 func (wf *WakuFilterFullNode) Stop() {
-	if wf.cancel == nil {
-		return
-	}
-
-	wf.h.RemoveStreamHandler(FilterSubscribeID_v20beta1)
-
-	wf.cancel()
-
-	wf.msgSub.Unsubscribe()
-
-	wf.wg.Wait()
+	wf.AppDesign.Stop(func() {
```

Comment:
I am finding this pattern hard to go through in the code. 

Inside Stop function, we are calling Stop for AppDesign to which we are passing another set of instructions to stop the original component (e.g: WakuFilterFullNode).  It is somehow looking convoluted.

Wondering if there is a simpler way of doing this which makes code easy to read.
One simple way could be to have a `stop` function which contains all of these actual stop instructions.

Reply:
I understood the importance of having this CommonService. Let me include a suggestion to simplify how the usage makes the code look.